README.md

```diff
@@ -331,6 +331,75 @@ needing to worry about token freshness:
 }];
 ```
 
+### Custom External User-Agents
+
+Each OAuth flow involves presenting an external user-agent to the user, that
+allows them to interact with the OAuth authorization server. Typical examples
+of a user-agent are the user's browser, or an incarnation of the browser like
+`ASWebAuthenticationSession` on iOS.
+
+AppAuth ships with several implementations of an external user-agent out of the
+box, including defaults for iOS and macOS that should be suitable for most
+cases. These default user-agents typically share persistent cookies with the
+system default browser, to improve the chance that the user doesn't need to
+sign-in all over again.
+
+It is possible to change the user-agent that AppAuth uses, and even write your
+own - all without needing to fork the library. All implementations of the
+external user-agent, be they included or created by you simply need to conform
+to the 
+[`OIDExternalUserAgent`](http://openid.github.io/AppAuth-iOS/docs/latest/protocol_o_i_d_external_user_agent-p.html)
+protocol.
+
+You then can pass instances of that into the 
+[`OIDAuthState.authStateByPresentingAuthorizationRequest:externalUserAgent:callback`](http://openid.github.io/AppAuth-iOS/docs/latest/interface_o_i_d_auth_state.html#ac762fe2bf95c116f0b437419be211fa1) method,
+rather than using the platform-specific convenience methods like 
+[`OIDAuthState.authStateByPresentingAuthorizationRequest:presentingViewController:callback:`](http://openid.github.io/AppAuth-iOS/docs/latest/category_o_i_d_auth_state_07_i_o_s_08.html#ae32fd0732cd3192cd5219f2655a4c85c)
+which creates the default user-agent.
+
+AppAuth for iOS includes a few extra user-agent implementations which you can
+try, for example the authorization request code in the above example can be rewritten as:
+
+<sub>Objective-C</sub>
```

Comment:
Maybe add a Swift version too?

Reply:
Would be good to have, I agree. My current priority is to get this info in the docs ASAP since it seems a lot of people need it.